### PR TITLE
sync loadbalancer status from host service

### LINKED ISSUE #732: Sync the .status.loadBalancer field of the mapped Services.
### Is your feature request related to a problem?

Services created by vcluster based on the [configured service mapping](https://www.vcluster.com/docs/architecture/networking#mapping-services-between-vcluster-and-host-cluster) are lacking status, which means that it is not possible to read for example the IP of the assigned load balancer.

### Which solution do you suggest?

Sync the `.status.loadBalancer` field of [the mapped Services](https://www.vcluster.com/docs/architecture/networking#mapping-services-between-vcluster-and-host-cluster).

### Which alternative solutions exist?

No viable solution. A workaround is to read the `.status.loadBalancer` from the source Service, but that requires additional knowledge, access, and manual effort from the user.

### Additional context

This is the controller that executes synchronization of the mapped Services:
[https://github.com/loft-sh/vcluster/blob/main/pkg/controllers/servicesync/servicesync.go](https://github.com/loft-sh/vcluster/blob/main/pkg/controllers/servicesync/servicesync.go)

### PR DESCRIPTION
**Pull request by rahulii**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind feature
**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #732 

**Please provide a short message that should be published in the vcluster release notes**
Services created by vcluster based on the configured service mapping are lacking status. This PR sync the loadbalancer status from host service to service created in vcluster

**What else do we need to know?**

### CODE REVIEW

File: pkg/controllers/servicesync/servicesync.go [hunk 1]
```diff
@@ -195,6 +196,12 @@ func (e *ServiceSyncer) syncServiceAndEndpoints(ctx context.Context, fromService
 		return ctrl.Result{}, nil
 	}
 
+	// sync the loadbalancer status
+	if fromService.Spec.Type == corev1.ServiceTypeLoadBalancer && !apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, corev1.LoadBalancerStatus{}) && !apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, toService.Status.LoadBalancer) {
```
Comment (matskiv) [inline pkg/controllers/servicesync/servicesync.go:200]: I am curious to know why is part of the condition needed?
```
!apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, corev1.LoadBalancerStatus{})
```

I would expect that if the state changes from having something in the `.Status.LoadBalancer` to having nothing, then we would like to have that reflected in the virtual Service resource too.

Comment (rahulii) [reply pkg/controllers/servicesync/servicesync.go:200]: ah, that makes sense!

**Comment by rahulii** [comment]
@matskiv do we also need tests for this?

**Review by matskiv — APPROVED** [review]
LGTM, thank you!

**Comment by matskiv** [comment]
@rahulii The tests would be nice, but since we don't have any tests set up for this controller yet, I don't think it should be mandatory.

**Comment by rahulii** [comment]
> this

cool! Thank for all the help

**Comment by rahulii** [comment]
> @rahulii The tests would be nice, but since we don't have any tests set up for this controller yet, I don't think it should be mandatory.

cool, thanks for all the help :)

**Comment by matskiv** [comment]
Hello @rahulii  :wave: 
We are planning to bring some vcluster T-shirts to KubeCon Europe, and I want to set some aside for the contributors that will be present at the conference. Are you attending by any chance? :)

**Pull request merged**